torchao/quantization/prototype/mixed_precision/scripts/BO_inference_speed.py

```diff
@@ -0,0 +1,524 @@
+import sys
+ 
+import torch
+import time
+import torch.nn as nn
+from torchao.quantization import quantize_
+import random
+from naive_intNwo import intN_weight_only
+
+import copy
+from lm_eval.evaluator import evaluate
+from lm_eval.models.huggingface import HFLM
+from lm_eval.tasks import get_task_dict
+
+from transformers import AutoModelForCausalLM, AutoTokenizer
+from ax.service.ax_client import AxClient, ObjectiveProperties
+import torch.multiprocessing as mp
+
+import os
+import sys
+from pathlib import Path
+from typing import Optional, Tuple
+from datetime import datetime
+import torchao
+import torch._dynamo.config
+import torch._inductor.config
+from torchao.utils import get_model_size_in_bytes
+from torchao.utils import TORCH_VERSION_AT_LEAST_2_5
+from torchao.quantization.quant_api import int4_weight_only
+from torchao._models.llama.model import Transformer, prepare_inputs_for_model
+from torchao._models.llama.tokenizer import get_tokenizer
+from torchao._models._eval import TransformerEvalWrapper, InputRecorder
+
+from torchao.dtypes import TensorCoreTiledLayoutType
+
+from torchao._models.llama.generate import ( 
+    device_sync,
+    multinomial_sample_one_no_sync,
+    logits_to_probs,
+    sample,
+    prefill,
+    decode_one_token,
+    model_forward,
+    encode_tokens,
+    _load_model,
+)
+
+default_device = 'cuda' if torch.cuda.is_available() else 'cpu'
+
+# quantize a model based on a given quantization configuration
+def quantize_by_fqn_to_config(model, device, fqn_to_config):
```

Comment:
Please refactor these two scripts to reuse as much code as possible! I think this (and many other functions) are duplicated right now

Reply:
made some of them into utils function